### PR TITLE
Update _getFormalRoleName to use the acl table

### PR DESCRIPTION
**Pull request by ryanrath**
## Description
- Just another incremental PR for converting XDUser over to acls. 
- Also reduced the lines of code where I could.

## Motivation and Context
To continue migrating XDUser to acls.

## Tests performed
Tests have been included for this function as well as functions that utilize this function. They include: 
- `getMostPrivilegedRole`
- `getAllRoles`
- `setActiveRole`
- `__construct` ( via the testCreateUser* functions )

These tests reside within `component_tests/lib/XDUserTest.php`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by jpwhite4** [comment]
The pull request description mentions that there are new tests - did you forget to git add the test file(s)?

**Comment by ryanrath** [comment]
I was actually just thinking about that this morning. I didn't forget per se, I was actually a little over zealous last week and added the tests for a great many of the functions that would be refactored. The ones that cover this function are:
- XDUserTests.php::testGetFormalRoleName
  - test cases include: 'cd', 'cs', 'pi', 'usr', and 'pub'
  - **NOTE:** There will be a slight change to the `dataProvider` function. I'll be adding entries for null / empty string while removing: `XDUserTest.php::testGetFormalRoleNameNull()` and `XDUserTest.php::testGetFormalRoleNameEmptyString`

**Comment by ryanrath** [comment]
@jpwhite4 My thought behind adding the tests first was to be able to assert that the tests passed before the changes.

File: classes/XDUser.php [hunk 1]
```diff
@@ -2551,23 +2551,19 @@ private function _formalizeTimestamp($db_timestamp)
...
     {
-
-        if ($role_abbrev == ROLE_ID_PUBLIC) {
-            return 'Public';
-        }
-
-        if ($role_abbrev == NULL) {
-            return 'Public';
+        $publicAclDisplay = 'Public';
+        if ($role_abbrev == ROLE_ID_PUBLIC || $role_abbrev == NULL) {
```
Comment (smgallo) [inline classes/XDUser.php]: `NULL` should be `null` as per style guide.

File: classes/XDUser.php [hunk 2]
```diff
@@ -2551,23 +2551,19 @@ private function _formalizeTimestamp($db_timestamp)
...
     public function _getFormalRoleName($role_abbrev)
     {
-
-        if ($role_abbrev == ROLE_ID_PUBLIC) {
-            return 'Public';
-        }
-
-        if ($role_abbrev == NULL) {
-            return 'Public';
+        $publicAclDisplay = 'Public';
```
Comment (smgallo) [inline classes/XDUser.php]: Why is `'Public'` hardcoded when there is already a public role in `moddb.Roles` (role_id = 9) and in the `moddb.acls` table (acl_id = 8)? Shouldn't this information be coming from the database? What if we update the database? Then we also need to update this function.

File: configuration/etl/etl_sql.d/acls/xdmod/create_public_user.sql [hunk 3]
```diff
@@ -22,6 +22,9 @@ LEFT JOIN Users cur
         BINARY cur.email_address = BINARY inc.email_address
 WHERE cur.id IS NULL;
 
+-- Update public acl display
+UPDATE acls SET display = 'Public' WHERE name = 'pub';
+
```
Comment (smgallo) [inline configuration/etl/etl_sql.d/acls/xdmod/create_public_user.sql]: Don't forget that we'll also need a migration as well.

Comment (ryanrath) [reply configuration/etl/etl_sql.d/acls/xdmod/create_public_user.sql]: Yup, we have one. That's a big reason why I included this change here as opposed to somewhere else. This sql file is executed as part of the Migration from 701to710.

Comment (smgallo) [reply configuration/etl/etl_sql.d/acls/xdmod/create_public_user.sql]: OK, sounds good.

**Pull request merged**